lib/foreman_tasks/cleaner.rb

```diff
@@ -90,13 +93,29 @@ def tasks
       ForemanTasks::Task.search_for(full_filter).select('DISTINCT foreman_tasks_tasks.id, foreman_tasks_tasks.type, foreman_tasks_tasks.external_id')
     end
 
-    def delete_tasks(chunk)
-      ForemanTasks::Task.where(:id => chunk.map(&:id)).delete_all
+    def delete_tasks(chunk, backup_dir = ForemanTasks.dynflow.world.persistence.current_backup_dir)
```

Comment:
This breaks compatibility with remote execution's extensions for the cleaner, I'm not against the change, just making a note of it